app/src/main/java/com/github/palFinderTeam/palfinder/meetups/fragments/MeetupViewFragment.kt

```diff
@@ -66,6 +69,80 @@ class MeetupViewFragment : Fragment() {
         }
 
         setTextView(R.id.tv_ViewStartDate, startDate)
-        setTextView(R.id.tv_ViewEndDate,endDate)
+        setTextView(R.id.tv_ViewEndDate, endDate)
+
+        setTextView(R.id.tv_ViewParticipants, getParticipation(meetUp))
+        setTextView(R.id.tv_ViewAge, getAgeRange(meetUp))
+        setTextView(R.id.tv_ViewGender, meetUp.criterionGender?.genderName.orEmpty())
+
+        view?.findViewById<TextView>(R.id.tv_ViewEventCreator)?.setOnClickListener { openProfile() }
+    }
+
+    /**
+     * Inject image or remove view if empty image
+     * @param meetUp
+     */
+    private fun injectImage(meetUp: MeetUp) {
+        viewModel.viewModelScope.launch {
+            if (meetUp.iconImage == null) {
+                requireView().findViewById<ImageView>(R.id.iv_MeetupImage).visibility = View.GONE
+            } else {
+                meetUp.iconImage.loadImageInto(
+                    requireView().findViewById(R.id.iv_MeetupImage),
+                    requireContext()
+                )
+            }
+        }
+    }
+
+    /**
+     * Creates the format for the age range
+     * @param meetUp
+     */
+    private fun getAgeRange(meetUp: MeetUp): String {
+        var marginStart = meetUp.criterionAge?.first.toString()
+        var marginEnd = meetUp.criterionAge?.second.toString()
+        if (meetUp.criterionAge?.second == Int.MAX_VALUE) {
+            // Display 66+ if second is MAX_INT
+            marginEnd = getString(R.string.meetup_view_age_max_plus)
+        }
+
+        return String.format(
+            getString(R.string.meetup_view_age),
+            marginStart,
+            marginEnd
+        )
+    }
+
+    /**
+     * Creates the format for the participates number
+     * @param meetUp
+     */
+    private fun getParticipation(meetUp: MeetUp): String {
+        return if (meetUp.hasMaxCapacity) {
+            // Limited participants
+            String.format(
+                getString(R.string.meetup_view_participation_limited),
+                meetUp.numberOfParticipants(),
+                meetUp.capacity
+            )
+        } else {
+            // Unlimited participants
+            String.format(
+                getString(R.string.meetup_view_participation_unlimited),
+                meetUp.numberOfParticipants()
+            )
+        }
     }
+
+    /**
+     * Onclick open profile user safely
+     */
+    private fun openProfile() {
```

Comment:
could you maybe allow to click on the user only if the meetUp is correctly fetched ? This could spare a crash if the user is offline or with low connection

Reply:
Instead of doing that, we could maybe create a general error page that the app enters if something wasn't correctly fetched, or is missing (profile, meetups,...)